src/interfaces/interfaces.ts

```diff
@@ -71,10 +71,18 @@ namespace interfaces {
     cache: null | TActivated | Promise<TActivated>;
   }
 
-  export type Factory<T, U extends unknown[] = unknown[], V extends unknown[] = unknown[]> = (...args: U) => (((...args: V) => T) | T);
+  export type SimpleFactory<T, U extends unknown[] = unknown[]> =  (...args: U) => T;
+
+  export type MultiFactory<T, U extends unknown[] = unknown[], V extends unknown[] = unknown[]> = (...args: U) => SimpleFactory<T,V>
```

Comment:
@PodaruDragos 
Helper types added.  We need a better name for MultiFactory

Reply:
We need some of the markdown files to be generated from actual tests so that required updates to markdown automatically occur when we change types.  Will also prevent incorrect typing that I encountered when adding new functionality.

Reply:
what about `CompoundFactory` ? 
I also think `MultiFactory` is fine